hydra_base/lib/units.py

```diff
@@ -295,13 +325,32 @@ def get_unit_dimension(measure_or_unit_abbreviation,**kwargs):
         dimension = db.DBSession.query(Dimension).filter(Dimension.id==units[0].dimension_id).one()
         return str(dimension.name)
 
+def get_dimension_by_unit_id(unit_id, do_accept_unit_id_none=False, **kwargs):
+    """
+        Return the physical dimension a given unit id refers to.
+        if do_raise is True, it raise an exception if unit_id is not valid or None
+        if do_raise is False, and unit_id is None, the function returns a Dimension with id None (unit_id can be none in some cases)
+    """
+    if do_accept_unit_id_none == True and unit_id is None:
+        # In this special case, the method returns a dimension with id None
+        return get_empty_dimension()
+
+    try:
+        dimension = db.DBSession.query(Dimension).join(Unit).filter(Unit.id==unit_id).filter().one()
+        return get_dimension(dimension.id)
+    except NoResultFound:
+        # The dimension does not exist
```

Comment:
do_raise is not used here. WHy is it in the doc string?

Reply:
I had to change it to the flag "do_accept_unit_id_none" that is more meangful in this case
Done it now